Polyfills/Window/Source/Window.cpp

```diff
@@ -32,46 +40,85 @@ namespace
             , m_thread{&TimeoutDispatcher::WaitThenCallProc, this}
         { }
 
-        void Dispatch(std::shared_ptr<Napi::FunctionReference> func, std::chrono::system_clock::time_point time) 
-        {   
-            if (time <= std::chrono::system_clock::now())
+        TimeoutId Dispatch(std::shared_ptr<Napi::FunctionReference> func, Milliseconds time)
+        {
+            if (time <= 0)
             {
                 CallFunction(std::move(func));
-                return;
+                return 0;
```

Comment:
Is this the correct behavior? `setTimeout(func, 0)` is supposed to call the `func` on the next frame. See the `Late timeouts` section from https://developer.mozilla.org/en-US/docs/Web/API/setTimeout.

Reply:
Also, setTimeout must return a positive number according to the spec.

Reply:
For late timeouts, how do we make sure they don't get dispatched until after all the currently executing code has been run?

Reply:
I think we have to use the WorkQueue.